test/metabase/query_processor_test/filter_test.clj

```diff
@@ -109,6 +109,30 @@
                    {:aggregation [[:count]]
                     :filter      [:between [:datetime-field $date :day] "2015-04-01" "2015-05-01"]}))))))))
 
+(deftest between-temporal-arithmetic-test
+  (testing "Should be able to use temporal arithmetic expressions in a `:between` filter (#22531)"
+    ;; we also want to test this against MongoDB but [[mt/normal-drivers-with-feature]] would normally not include that
+    ;; since MongoDB only supports expressions if version is 4.0 or above and [[mt/normal-drivers-with-feature]]
+    ;; currently uses [[driver/supports?]] rather than [[driver/database-supports?]] (TODO FIXME)
+    (mt/with-clock #t "2022-06-10T16:12-08:00[US/Pacific]"
```

Comment:
`with-clock` does not affect the clocks of the Databases we connect to so it's not going to affect relative datetimes or anything like that. If this test passes now it will start breaking at some point in the future

Reply:
Will reverse.

Reply:
If you want to make sure that we actually get the correct rows you can take a look at the tests in https://github.com/metabase/metabase/blob/master/test/metabase/query_processor_test/date_bucketing_test.clj which actually create a new dataset on the fly to run tests against so we can verify that relative datetime things actually return the correct rows. You can use those without using `with-clock`